Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/connected_components.h

```diff
@@ -931,23 +931,48 @@ struct No_mark
 
 template < class PolygonMesh, class PolygonMeshRange,
            class FIMap, class VIMap,
-           class HIMap, class Ecm >
+           class HIMap, class Ecm, class NamedParameters >
 void split_connected_components_impl(FIMap fim,
                                      HIMap him,
                                      VIMap vim,
                                      Ecm ecm,
                                      PolygonMeshRange& range,
-                                     const PolygonMesh& tm)
+                                     const PolygonMesh& tm,
+                                     const NamedParameters& np)
 {
-  typename boost::template property_map<
+  typedef typename internal_np::Lookup_named_param_def <
+      internal_np::face_patch_t,
+      NamedParameters,
+      typename boost::template property_map<
       PolygonMesh, CGAL::dynamic_face_property_t<int > >::const_type
-      pidmap = get(CGAL::dynamic_face_property_t<int>(), tm);
+      > ::type Fpm;
+
+  using parameters::choose_parameter;
+  using parameters::get_parameter;
+  using parameters::is_default_parameter;
+
+  Fpm pidmap = choose_parameter(get_parameter(np, internal_np::face_patch),
+                                get(CGAL::dynamic_face_property_t<int>(), tm));
```

Comment:
the face component map of PMP::CC must have value type `boost::graph_traits<PolygonMesh>::faces_size_type`